tests/test_heterobatch.py

```diff
@@ -0,0 +1,441 @@
+"""
+Test module for heterobatch implementation of Lomb-Scargle periodogram.
+Compares heterobatch results against single-batch calculations.
+"""
+
+from __future__ import annotations
+
+from functools import partial
+
+import numpy as np
+import pytest
+
+import nifty_ls
+import nifty_ls.backends
+from nifty_ls.test_helpers.utils import gen_data_mp
+
+
+@pytest.fixture
+def data(request):
+    """Generate test data with specified N data points and N_batch series"""
+    # Check if this is a parametrized call
+    if hasattr(request, 'param'):
+        N_series = request.param.get('N_series', 10_000)
+        N_d = request.param.get('N', 100)
+        N_batch = request.param.get('N_batch', 10)
+    else:
+        # Default values when not parametrized
+        N_series = 100
+        N_d = 50
+        N_batch = 3
+
+    return gen_data_mp(N_series=N_series, N_d=N_d, N_batch=N_batch)
+
+
+@pytest.fixture(scope='module')
+def nifty_backend(request):
+    """Set up the backend function based on the parameterized backend name"""
+    available_heterobatch_backends = nifty_ls.backends.available_backends(
+        backend_names=nifty_ls.backends.HETEROBATCH_BACKEND_NAMES
+    )
+
+    if request.param in available_heterobatch_backends:
+        return (
+            request.param,
+            partial(nifty_ls.lombscargle_heterobatch, backend=request.param),
+        )
+    else:
+        pytest.skip(f'Backend {request.param} is not available')
+
+
+@pytest.mark.parametrize('Nf', [1_000, 10_000])
+@pytest.mark.parametrize(
+    'data',
+    [
+        {'N_series': 1000, 'N': 1000, 'N_batch': 1},
+        {'N_series': 1000, 'N': 100, 'N_batch': 5},
+        {'N_series': 10_000, 'N': 100, 'N_batch': 1},
+    ],
+    indirect=['data'],
+)
```

Comment:
The heterobatch tests are pretty slow, maybe we can test just `Nf = 1000`, and do a smaller N_series, like 100? I think testing batched and non-batched is good, although we probably don't need two non-batched tests.